pysb/tests/test_exporters.py

```diff
@@ -101,6 +106,13 @@ def check_convert(model, format):
                     print(pd.DataFrame(dict(rr=rr_obs, pysb=py_obs)))
                     raise ValueError('Model {}, observable__o{} "{}" trajectories do not match:'.format(
                         model.name, obs_idx, model.observables[obs_idx].name))
+        elif format == 'json':
+            # Round-trip the model by re-importing the JSON
+            m = model_from_json(exported_file)
+            # Check network generation and single-step integrator
+            if model.name not in ('pysb.examples.tutorial_b',
+                                  'pysb.examples.tutorial_c'):
+                ScipyOdeSimulator(m)
```

Comment:
This is a reasonable smoke test to make sure the model isn't broken, but it doesn't demonstrate we got the same model back. We ought to actually check that the model is the same, but that's probably infeasible on every example unless we leverage some other exporter (and hope we don't end up with parallel bugs in both exporters). The first thing that comes to mind is to run each model and its re-imported version through the pysb_flat exporter and compare the strings. Of course the pysb_flat exporter doesn't have a round-trip fidelity test either, does it?

Or maybe create a minimal model that exercises every feature once, and explicitly check all the components before and after re-import? That might be a big job, though.

Reply:
Indeed - I used `PySBFlatExporter` to manually check things were working as expected, but symbols within expressions don't appear in a deterministic order. I agree the best solution is probably to create a toy model that uses every feature once and check equality manually.

Reply:
I ended up using `check_model_against_component_list` on all of the example models after round-tripping to JSON, which seems to do the job.